source/swap/docs/Swap.md

```diff
@@ -0,0 +1,197 @@
+
+
+### Functions
+
+### <span id="Swap-constructor--"></span> `constructor()` (public)
+
+Contract Constructor
+
+
+Sets domain for signature validation (EIP-712)
+
+
+
+
+### <span id="Swap-swap-struct-Types-Order-"></span> `swap(struct Types.Order order)` (external)
+
+Atomic Token Swap
+
+
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`order`|Types.Order Order to settle
+
+
+
+### <span id="Swap-cancel-uint256---"></span> `cancel(uint256[] nonces)` (external)
+
+Cancel one or more open orders by nonce
+
+
+Cancelled nonces are marked UNAVAILABLE (0x01)
+Emits a Cancel event
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`nonces`|uint256[] List of nonces to cancel
+
+
+
+### <span id="Swap-invalidate-uint256-"></span> `invalidate(uint256 minimumNonce)` (external)
+
+Invalidate all orders below a nonce value
+
+
+Emits an Invalidate event
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`minimumNonce`|uint256 Minimum valid nonce
+
+
+
+### <span id="Swap-authorizeSender-address-"></span> `authorizeSender(address authorizedSender)` (external)
+
+Authorize a delegated sender
+
+
+Emits an AuthorizeSender event
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizedSender`|address Address to authorize
+
+
+
+### <span id="Swap-authorizeSigner-address-"></span> `authorizeSigner(address authorizedSigner)` (external)
+
+Authorize a delegated signer
+
+
+Emits an AuthorizeSigner event
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizedSigner`|address Address to authorize
+
+
+
+### <span id="Swap-revokeSender-address-"></span> `revokeSender(address authorizedSender)` (external)
+
+Revoke an authorized sender
+
+
+Emits a RevokeSender event
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizedSender`|address Address to revoke
+
+
+
+### <span id="Swap-revokeSigner-address-"></span> `revokeSigner(address authorizedSigner)` (external)
+
+Revoke an authorized signer
+
+
+Emits a RevokeSigner event
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizedSigner`|address Address to revoke
+
+
+
+### <span id="Swap-isSenderAuthorized-address-address-"></span> `isSenderAuthorized(address authorizer, address delegate) → bool` (internal)
+
+Determine whether a sender delegate is authorized
+
+
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizer`|address Address doing the authorization
+|`delegate`|address Address being authorized
+
+
+|  Param    | Description   |
+|   ---------| :------------ |
+| `bool`|  True if a delegate is authorized to send
```

Comment:
Can we add a title or a line to make it clear that this is the return values table? It looks the exact same as the parameters